### ISSUE TITLE
bug: Custom snippets not working after latest update

### ISSUE THREAD

**Issue by ianchesal** [LazyVim/LazyVim#3240]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

v0.10.0

### Operating system/version

MacOS 14.5

### Describe the bug

I have custom snippets in my config path at `snippets/gitcommit.json` -- these were working with LuaSnip.

With the update to Neovim 0.10 and LazyVim's switch to native Neovim snippets these no longer work.

I do see that nvim-snippet and friendly-snippets get loaded by Lazy when I start typing in a git commit. `friendly-snippet` snippets actually work -- I type `feat` and nvim-cmp shows me the canonical commit snippet. But custom snippets from that file are not available to me.

`vim.fn.stdpath('config')` prints the expected path to my configuration and I see my `snippets/gitcommit.json` file at that path.

Snippets file is here: https://github.com/ianchesal/dotfiles/blob/main/nvim/snippets/gitcommit.json

My configuration is here: https://github.com/ianchesal/dotfiles/tree/main/nvim

I've been over the docs and the LazyVim configs and I can't see why this isn't being loaded.

### Steps To Reproduce

1. Place my [gitcommit.json](https://github.com/ianchesal/dotfiles/blob/main/nvim/snippets/gitcommit.json) file at `snippets/gitcommit.json`
2. Open a git commit with neovim as your editor
3. Type `risk` -- you see no nvim-cmp completions when you type this.

<img width="658" alt="Screenshot 2024-05-19 at 9 19 35 AM" src="https://github.com/LazyVim/LazyVim/assets/291808/554777f4-7746-4081-8222-a6921007132a">

Screen shot shows that `ris` produces no nvim-cmp completions to choose from when in a git commit.

### Expected Behavior

Expected behavior is `nvim-cmp` shows you two possible, custom snippets you can choose from based on the contents of that custom snippet file.

### Repro

```Lua
I'm not sure how to repo this with pure lua, sorry.
```

**Comment by ianchesal** [comment]
I forgot to note: if I re-enable `luasnip` the custom snippets work. Here's what working looks like:

<img width="615" alt="Screenshot 2024-05-19 at 9 22 10 AM" src="https://github.com/LazyVim/LazyVim/assets/291808/e60e0f90-815e-43d7-8c43-08667024db5b">

This could possibly be a bug in `vim-snippets` as well.

**Comment by dpetka2001** [comment]
You have to configure `nvim-snippets`, see [docs](https://github.com/garymjr/nvim-snippets?tab=readme-ov-file#configuration) for `search_paths`.

**Comment by folke** [comment]
Custom snippets may require some additional configuration.
See here https://github.com/garymjr/nvim-snippets

**Comment by ianchesal** [comment]
Thanks @folke -- I read over that before filing. The default for  `search_paths` should cover this. But I'll set that explicitly. I appreciate all the recent updates! LazyVim is in an amazing spot.

**Comment by ianchesal** [comment]
Try as I might, I could not find a set of `nvim-snippets` configuration options that work. Any combination of settings for `nvim-snippets` ends up disabling `friendly-snippet` completion and does not fix the problem with custom snippets not working.

There's some conflux of `nvim-snippets` configuration and LavyVim configuration that I think might be intersecting here and causing the trouble.

I wrote some notes about what I tried in this thread: https://old.reddit.com/r/neovim/comments/1cw9jt3/using_native_snippets_in_lazyvim/l4vgdp4/

I have to put it down for now. I'll pick it back up again tonight after work.

**Comment by dpetka2001** [comment]
@ianchesal 
Can you try adding the following to your personal configuration
```lua
-- ~/.config/nvim/lua/plugins/snip.lua
return {
      {
        "garymjr/nvim-snippets",
        dependencies = {
          "rafamadriz/friendly-snippets",
        },
        opts = { search_paths = { vim.fn.stdpath("config") .. "/misc/snippets" } },
      },
}
```
For the custom snippets, pls see [this comment](https://github.com/LazyVim/LazyVim/discussions/3216#discussioncomment-9479930). It seems the plugin does not handle nested folders. So, like that comment suggests, I have all my snippets under `/misc/snippets/`.

**Comment by rvdwijngaard** [comment]
@dpetka2001 I tried that but that does not seem to help.

**Comment by rvdwijngaard** [comment]
I did some further testing and I noticed indeed that custom snippets do not work. Other [global](https://github.com/rafamadriz/friendly-snippets/blob/main/snippets/global.json)  snippets from `friendly-snippets` are also not working. For example `dateYMD` or `uuid`. Seems like the same problem as reported in this issue. File specific snippets are available though. 

If I enable luasnip everything works as expected.

**Comment by dpetka2001** [comment]
You need to enable `global_snippets = { "all", "global" },` in the `opts` for `global` to be included.

**Comment by dpetka2001** [comment]
I created a test repo with default LazyVim settings and then added native snippets support. You can see the changes from the default LazyVim installation in this [commit](https://github.com/dpetka2001/test/commit/fbce434eb382222e02f0a23e0e6bede23348c535). Nested folders work for me for custom snippets, as well as friendly-snippets tested in Python and Lua files.

**Comment by folke** [comment]
https://github.com/garymjr/nvim-snippets/pull/10

**Comment by folke** [comment]
I've linked nvim-snippets to my PR for now till they merge it.

**Comment by phuongthuan** [comment]
I'm using luasnip + nvim-cmp and get the same issue in nvim 0.10
Has anyone moved to `nvim-snippets` yet? or still sticking with `luasnip`?

**Comment by folke** [comment]
My PR has been merged, so custom snippets work on 0.10.
You say you're using luasnip. This is not about luasnip.
You cant use both at the same time
